docs/api/tensorflow_usage.rst

```diff
@@ -0,0 +1,55 @@
+:desc: Find out how to configure your environment for efficient usage of TensorFlow inside Rasa Open Source
+
+.. _tensorflow_usage:
+
+Setting up the TensorFlow Runtime
+=================================
+
+TensorFlow allows setting the runtime environment via
+`TF Config submodule <https://www.tensorflow.org/api_docs/python/tf/config>`_. Rasa Open Source supports a smaller subset of these
+configuration options and makes appropriate calls to the ``tf.config`` submodule.
+This smaller subset comprises of configurations that developers frequently use with Rasa Open Source.
+All configuration options are specified using environment variables as shown in subsequent sections.
+
+Optimizing CPU Performance
+--------------------------
+
+Parallelizing one operation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set ``TF_INTRA_OP_PARALLELISM_THREADS`` as an environment variable to specify maximum number of threads that can be used
+to parallelize the execution of one operation. If left unspecified, this value defaults to ``0`` which means TensorFlow should
+pick an appropriate value depending on the system configuration.
+
+
+Parallelizing multiple operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Set ``TF_INTER_OP_PARALLELISM_THREADS`` as an environment variable to specify maximum number of threads that can be used
+to parallelize the execution of multiple **non-blocking** operations. If left unspecified, this value defaults to ``0``
+which means TensorFlow should pick an appropriate value depending on the system configuration.
+
+
+Optimizing GPU Performance
+--------------------------
+
+Limiting GPU memory growth
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TensorFlow by default blocks all the available GPU memory for the running process. This can be limiting if you are running
+multiple TensorFlow processes and want to distribute memory across them. To prevent this,
+set an environment variable ``TF_FORCE_GPU_ALLOW_GROWTH`` to ``True``.
```

Comment:
`TF_FORCE_GPU_ALLOW_GROWTH` means the gpu is forced to grow to use more memory than available or how can is understand that? Or should this be set to `False`?

Reply:
@wochinge This is an environment variable that Tensorflow provides and I agree that unfortunately it isn't coined very well. So, if the value is `False`(default value), then the process blocks the complete GPU memory available. If it is `True` it would allocate memory as required and would **grow** only if required. In no way, the process that use more memory than the GPU provides it.

Reply:
Wild 😄